spec/spec_helper.rb

```diff
@@ -94,3 +94,60 @@
   Kernel.srand config.seed
 =end
 end
+
+##
+# This custom matcher allows to write Rspec expectations using mocks with
+# the following idiom:
+#
+#     expect {
+#       code_to_test_here
+#     } .to satisfy proc {
+#       mocks expectations here
+#     }
```

Comment:
To make specs using mocks to have the same shape as the other specs where the expectations are usually at the end of the test.

I did not get from the docs that the ordering was a convention for the use of mocks, I will roll it back then.

Reply:
With convention, I mean that is the usual way to use it, whenever a new developer comes and sees this will have to check here what is this, and why was it implemented.
In this way we are adding a new method which does not give any functionality at all, it only modifies the way that the things are done.